### ISSUE TITLE
False Positive Error

### ISSUE THREAD

**Issue by Tomperez98** [charliermarsh/ruff#1229]
```python
"""Example code."""
import result


def some_function(first: int, second: int) -> result.Result[float, Exception]:
    """We use this function as an example for some real logic."""
    if second == 0:
        return result.Err(ZeroDivisionError())
    elif second >= 10:
        return result.Err(ValueError("Value is to big"))
    return result.Ok(first / second)
```

ruff detects:
```bash
sample/example.py:7:5: RET505 Unnecessary `elif` after `return` statement
```

This seems to be wrong

**Comment by Tomperez98** [comment]
This solves the ruff error detected

```python
"""Example code."""
import result


def some_function(first: int, second: int) -> result.Result[float, Exception]:
    """We use this function as an example for some real logic."""
    if second == 0:
        return result.Err(ZeroDivisionError())

    if second >= 10:
        return result.Err(ValueError("Value is to big"))

    return result.Ok(first / second)
```

Is this the expected behavior? Maybe it is, and the issue can be closed

**Comment by charliermarsh** [comment]
Thanks for this, appreciate the issue and the clear repo. I believe this is the rule “working as intended” so gonna close out. (I’m not a huge fan of those specific rules around unnecessary else blocks.)
